internal/testutil/db.go

```diff
@@ -40,9 +42,18 @@ func (d *DbTestConnector) Connect(ctx context.Context) (pgx.Tx, db.CloseFunc, er
 	if err != nil {
 		return nil, nil, err
 	}
+	d.innerTx = innerTx
 	return innerTx, func(context.Context) {}, nil
 }
 
+func (d *DbTestConnector) Exec(ctx context.Context, sql string, args ...interface{}) (pgconn.CommandTag, error) {
+	if d.innerTx != nil {
+		return d.innerTx.Exec(ctx, sql, args...)
```

Comment:
I kinda assumed that the reason the test connector was separate was so that all tests were wrapped in a transaction that we could revert at the end right? So it has to use the transaction, and not just any old connection? I tried with `d.tx.Exec()` but that panic'd, I think because pgx doesn't model nested transactions as a tree of children, but instead a stack, and if you try to use anything other than the top item in the stack, its an error. Thats how postgres actually works under the hood as well -- it's just savepoints within the same transaction as opposed to independent chunks that can evolve separately. Is that going to create test fuckery though? I dont know the codebase well enough to say.

Reply:
> I kinda assumed that the reason the test connector was separate was so that all tests were wrapped in a transaction that we could revert at the end right?

Yeah that's right. Since we rollback `d.tx` after each test I thought it would be fine.

> I tried with d.tx.Exec() but that panic'd, I think because pgx doesn't model nested transactions as a tree of children, but instead a stack, and if you try to use anything other than the top item in the stack

But yeah I didn't realize it wasn't actually a tree structure. Just played around with this and you're right. :+1: